tests/utils.py

```diff
@@ -143,3 +145,19 @@ def run_pattern(pattern, data, metadata):
     reverse_transformed = pattern.reverse_transform(transformed)
 
     return updated_metadata, transformed, reverse_transformed
+
+
+def run_copula(data, metadata, patterns):
```

Comment:
we should rename these to include the fact that they add constraints since they are in the general test utils. Other option is to make it optional to add the constraints